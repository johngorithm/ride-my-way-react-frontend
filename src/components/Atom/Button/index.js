import React,  {Fragment} from 'react';


const Button = ({ text, classes, ...props}) => {
  return (
    <Fragment>
      <button {...props} className={classes}>{text}</button>
    </Fragment>
  )
}
export default Button;
