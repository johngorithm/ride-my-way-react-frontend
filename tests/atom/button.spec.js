import React from 'react';
import { shallow } from 'enzyme';

import Button from 'components/Atom/Button';

const props = {
  text: 'SAVE',
  classes: 'button button primary'
}

describe('Button Tests', () => {
  test('it renders successfully', () => {
    const wrapper = shallow(<Button {...props} />);

    expect(wrapper.exists()).toBe(true);
  })

  test('it matches snapshot', () => {
    const wrapper = shallow(<Button {...props} />);

    expect(wrapper).toMatchSnapshot();
  })
})