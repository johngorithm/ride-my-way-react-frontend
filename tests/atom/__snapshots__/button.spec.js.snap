// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests it matches snapshot 1`] = `
<Fragment>
  <button
    className="button button primary"
  >
    SAVE
  </button>
</Fragment>
`;
